app/screens/server/index.tsx

```diff
@@ -1,105 +1,383 @@
 // Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
 // See LICENSE.txt for license information.
 
-import {Colors, DebugInstructions, LearnMoreLinks, ReloadInstructions} from 'react-native/Libraries/NewAppScreen';
-import {SafeAreaView, ScrollView, StatusBar, StyleSheet, Text, View} from 'react-native';
+import {ManagedConfig, useManagedConfig} from '@mattermost/react-native-emm';
+import React, {useCallback, useEffect, useRef, useState} from 'react';
+import {useIntl} from 'react-intl';
+import {
+    ActivityIndicator, EventSubscription, Image, Keyboard, KeyboardAvoidingView,
+    Platform, StatusBar, StyleSheet, TextInput, TouchableWithoutFeedback, View,
+} from 'react-native';
+import Button from 'react-native-button';
+import {NavigationFunctionComponent} from 'react-native-navigation';
+import {SafeAreaView} from 'react-native-safe-area-context';
 
-import React from 'react';
+import LocalConfig from '@assets/config.json';
+import AppVersion from '@components/app_version';
+import ErrorText, {ClientErrorWithIntl} from '@components/error_text';
+import FormattedText from '@components/formatted_text';
 import {Screens} from '@constants';
+import {doPing, fetchConfigAndLicense} from '@requests/remote/general';
 import {goToScreen} from '@screens/navigation';
+import {isMinimumServerVersion} from '@utils/helpers';
+import {preventDoubleTap} from '@utils/tap';
+import {changeOpacity, makeStyleSheetFromTheme} from '@utils/theme';
+import {getServerUrlAfterRedirect, isValidUrl, sanitizeUrl} from '@utils/url';
 
-declare const global: { HermesInternal: null | {} };
+import {DeepLinkWithData, LaunchProps, LaunchType} from '@typings/launch';
+
+interface ServerProps extends LaunchProps {
+    componentId: string;
+    theme: Theme;
+}
+
+let cancelPing: undefined | (() => void);
+
+const Server: NavigationFunctionComponent = ({extra, launchType, launchError, theme}: ServerProps) => {
+    // TODO: If we have LaunchProps, ensure they get passed along to subsequent screens
+    // so that they are eventually accessible in the Channel screen.
+
+    const intl = useIntl();
+    const managedConfig = useManagedConfig<ManagedConfig>();
```

Comment:
for the screen components, this could just be using the prop for `managedConfig` instead of using the hook. Both of the solutions would work just fine